cmd/src/users_clean.go

```diff
@@ -0,0 +1,202 @@
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			fmt.Printf("failed to parse command args with error: %s", err)
+		}
+
+		ctx := context.Background()
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+
+		vars := map[string]interface{}{
+			"-d": api.NullInt(*daysToDelete),
+		}
+
+		query := `
+query Users($first: Int, $query: String) {
+	users(first: $first, query: $query) {
+		nodes {
+			...UserFields
+		}
+	}
+}
+` + userFragment
+
+		// get users to delete
+		var result struct {
+			Users struct {
+				Nodes []User
+			}
+		}
+		if ok, err := client.NewRequest(query, vars).Do(ctx, &result); err != nil || !ok {
+			return err
+		}
+
+		usersToDelete := make([]UserToDelete, 0)
+		for _, user := range result.Users.Nodes {
+			daysSinceLastUse, wasLastActive, err := computeDaysSinceLastUse(user)
+			if err != nil {
+				return err
+			}
+			if !wasLastActive && !*removeNoLastActive {
+				continue
+			}
+			if *noAdmin && user.SiteAdmin {
+				continue
+			}
+			if daysSinceLastUse <= *daysToDelete && wasLastActive {
+				continue
+			}
+			deleteUser := UserToDelete{user, daysSinceLastUse}
+
+			usersToDelete = append(usersToDelete, deleteUser)
+			//fmt.Printf("\nAdding %s to remove list: %d days since last active, remove after %d days inactive\n", user.Username, daysSinceLastUse, *daysToDelete)
+		}
+
+		if *skipConfirmation {
+			for _, user := range usersToDelete {
+				if err := removeUser(user.User, client, ctx); err != nil {
+					return err
+				}
+				// TODO: see flag toEmail
+				//if *toEmail {
+				//	if err := sendEmail(user.User); err != nil {
+				//		fmt.Printf("failed to send email to %s with error: %s", user.User.Emails[0].Email, err)
+				//	}
+				//}
+			}
+			return nil
+		}
+
+		// confirm and remove users
+		if confirmed, _ := confirmUserRemoval(usersToDelete); !confirmed {
+			fmt.Println("Aborting removal")
+			return nil
+		} else {
+			fmt.Println("REMOVING USERS")
+			for _, user := range usersToDelete {
+				if err := removeUser(user.User, client, ctx); err != nil {
+					return err
+				}
+				// TODO: see flag toEmail
+				//if *toEmail {
+				//	if err := sendEmail(user.User); err != nil {
+				//		fmt.Printf("failed to send email to %s with error: %s", user.User.Emails[0].Email, err)
+				//	}
+				//}
+			}
+		}
+
+		return nil
+	}
+
+	// Register the command.
+	usersCommands = append(usersCommands, &command{
+		flagSet:   flagSet,
+		handler:   handler,
+		usageFunc: usageFunc,
+	})
+}
+
+func computeDaysSinceLastUse(user User) (timeDiff int, wasLastActive bool, _ error) {
+	timeNow := time.Now()
+	// handle for null lastActiveTime returned from
+	if user.UsageStatistics.LastActiveTime == "" {
+		wasLastActive = false
+		return 0, wasLastActive, nil
+	}
+	timeLast, err := time.Parse(time.RFC3339, user.UsageStatistics.LastActiveTime)
+	if err != nil {
+		fmt.Printf("failed to parse lastActive time: %s", err)
+		return 0, false, err
```

Comment:
We should either return or only print error, unless there is very good reason to do both.

Reply:
Removing the print error, this was mostly just for debugging null values in a users `lastActiveTime`